objects/osuToken.py

```diff
@@ -178,8 +178,11 @@ async def create_token(
     async with glob.redis.pipeline() as pipe:
         await pipe.sadd("bancho:tokens", token_id)
         await pipe.hset("bancho:tokens:json", token_id, orjson.dumps(token))
-        await pipe.set(f"bancho:tokens:ids:{user_id}", token_id)
-        await pipe.set(f"bancho:tokens:names:{safe_name}", token_id)
+        if not tournament:
+            await pipe.set(f"bancho:tokens:ids:{user_id}:primary", token_id)
+            await pipe.set(f"bancho:tokens:names:{safe_name}:primary", token_id)
+        await pipe.sadd(f"bancho:tokens:ids:{user_id}", token_id)
+        await pipe.sadd(f"bancho:tokens:names:{safe_name}", token_id)
```

Comment:
1. The `ids:` and `names:` indices have been changed from one-to-one, to one-to-many
2. New one-to-one `ids:{id}:primary` and `names:{safe_name}:primary` keys have been added to track the primary session. Previous behaviour for `ids:` and `names:` was to track the _latest_ login, which was usually undesirable/incorrect.

Reply:
I'm going to need to delete all session data to deploy this, to avoid ops against the wrong data types

Reply:
Note the implications of :primary not existing for all users -- lookups like `get_primary_token_by_user_id` will return None, EVEN IF there are non-primary sessions